api/dailies.ts

```diff
@@ -17,37 +19,90 @@ export function makeDailyWebhookPostHandler(
   return async function handlePostDailyWebhook(
     request: router.RouterRequest,
   ): Promise<Response> {
+    // Check the webhook token.
     const token = request.params["token"];
     if (webhookToken && token !== webhookToken) {
       return new Response("Invalid token", { status: 401 });
     }
 
-    return await handleExecuteDailyWebhook(lcClient, webhookURL);
+    // Get the season ID if applicable.
+    const seasonID = request.url.searchParams.get("season_id") || undefined;
+
+    // Execute the webhook.
+    return await executeDailyWebhook(
+      lcClient,
+      leaderboardClient,
+      webhookURL,
+      seasonID,
+    );
   };
 }
 
-async function handleExecuteDailyWebhook(
+async function executeDailyWebhook(
   lcClient: lc.LCClient,
+  leaderboardClient: leaderboard.LeaderboardClient,
   webhookURL: string,
+  seasonID?: string,
 ): Promise<Response> {
+  // Get the daily question.
   const question = await lcClient.getDailyQuestion();
-  const content = formatLCDailyQuestion(question);
+
+  // Get the season data if a season ID is provided or if it is Sunday.
+  const isSunday = new Date(question.date).getDay() === 0;
+  const season = seasonID
+    ? await leaderboardClient.getSeason(seasonID)
+    : isSunday
+    ? await leaderboardClient.getCurrentSeason()
+    : null;
+
+  // Format the webhook content.
+  const content = formatDailyWebhook({ question, season });
+
+  // Execute the webhook.
   await discord.executeWebhook({
     url: webhookURL,
     data: { content },
   });
 
+  // Acknowledge the request.
   return new Response("OK");
 }
 
-function formatLCDailyQuestion(question: lc.DailyQuestion): string {
-  return [
-    `## Daily Leetcode Question for ${question.date}`,
-    `**Question**: ${question.title}`,
-    `**Difficulty**: ${question.difficulty}`,
-    `**Link**: <${question.url}>`,
+/**
+ * FormatDailyWebhookOptions are the options for formatDailyWebhook.
+ */
+export interface FormatDailyWebhookOptions {
+  /**
+   * question is the daily question.
+   */
+  question: lc.DailyQuestion;
+
+  /**
+   * season is the season to recap.
+   */
+  season: leaderboard.Season | null;
+}
+
+/**
+ * formatDailyWebhook formats a daily webhook.
+ */
+export function formatDailyWebhook(
+  options: FormatDailyWebhookOptions,
+): string {
+  const content = [
+    `## Daily Leetcode Question for ${options.question.date}`,
+    `**Question**: ${options.question.title}`,
+    `**Difficulty**: ${options.question.difficulty}`,
+    `**Link**: <${options.question.url}>`,
     `**Snack**: Here is a snack to get your brain working: ${snacks.pickRandom()}`,
     "",
     "Submit your solution by typing `/lc submit YOUR_SUBMISSION_URL` below! ([more info](https://acmcsuf.com/lc-dailies-handbook))",
   ].join("\n");
+
+  if (!options.season) {
+    return content;
+  }
+
+  // TODO: Render the leaderboard.
```

Comment:
Resolve TODO comment.

Reply:
Resolved in <https://github.com/acmcsufoss/lc-dailies/pull/33/commits/4827940a2de731296b3b38d0affdbfeda8338b30>.